### ISSUE TITLE
react-native withOAuth logins not persisted after app restart

### ISSUE THREAD

**Issue by ejdoh1** [aws-amplify/amplify-js#4417]
**Describe the bug**
After a fresh create react native app using RN 0.61.4 and using App.js and aws-exports.js from (known working) https://github.com/manueliglesias/hosted-ui-react-native-cli, the user does not remain signed in after the app is restarted.

**To Reproduce**

1. Run the react native app from repo https://github.com/ejdoh1/rnhui
2. react-native run-android  
3. Launch the app and login using Google or other
4. After successful login, swipe-kill the app
5. Relaunch the app noting the user is prompted to re-login

**Expected behavior**
As with the [example repo](https://github.com/manueliglesias/hosted-ui-react-native-cli) using RN version 0.58, after the app is relaunched, the user should not be prompted to sign in again.

<details>
  <summary><strong>Environment</strong></summary>

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 115.46 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.2 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.13.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 78.0.3904.97
    Firefox: 69.0.1
    Safari: 13.0.1
  npmPackages:
    @babel/core: ^7.7.2 => 7.7.2 
    @babel/runtime: ^7.7.2 => 7.7.2 
    @react-native-community/eslint-config: ^0.0.5 => 0.0.5 
    aws-amplify: ^1.2.4 => 1.2.4 
    aws-amplify-react-native: ^2.2.3 => 2.2.3 
    babel-jest: ^24.9.0 => 24.9.0 
    eslint: ^6.6.0 => 6.6.0 
    jest: ^24.9.0 => 24.9.0 
    metro-react-native-babel-preset: ^0.57.0 => 0.57.0 
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.4 => 0.61.4 
    react-test-renderer: 16.9.0 => 16.9.0 
  npmGlobalPackages:
    @aws-amplify/cli: 3.9.0
    aws-sdk: 2.553.0
    azure-functions-core-tools: 2.7.1724
    expo-cli: 3.0.10
    ios-deploy: 1.9.4
    npm-update-all: 1.0.1
    npm: 6.13.0
    powerbi-visuals-tools: 3.1.5
    react-devtools: 3.6.3
    react-native-cli: 2.0.1
    react-native-rename: 2.4.1
    serverless-azure-functions: 1.0.0
    serverless: 1.57.0
```

</details>

**Smartphone (please complete the following information):**

- Device: Google Pixel 2
- OS: Android 9

**Sample code**
https://github.com/ejdoh1/rnhui  

**Logs**
After app re-launch...
```
[Log] [DEBUG] 06:11.274 AuthClass - getting current authenticated user (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129117)
[Log] [DEBUG] 06:11.275 AuthClass - cannot load federated user from auth storage (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129117)
[Log] [DEBUG] 06:11.275 AuthClass - get current authenticated userpool user (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129117)
[Log] [DEBUG] 06:11.276 AuthClass - Failed to get user from user pool (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129117)
[Log] [DEBUG] 06:11.276 AuthClass - The user is not authenticated by the error – "No current user" (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129127)
[Log] [DEBUG] 06:11.277 withOAuth - not authenticated (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129117)
[Log] [DEBUG] 06:11.357 urlListener - urlListener – {url: null} (blob:http://localhost:8081/79e78c06-d02a-4029-945b-c9b21ccf63af, line 129127)
```

**Comment by anthonyhumphreys** [comment]
#4351

**Comment by Ashish-Nanda** [comment]
Hi @ejdoh1 . Are you still facing this issue?
The fix for this problem has been available since version 2.2.0
